### PR TITLE
Fix menu behaviour after deleting item

### LINKED ISSUE #2501: Menus are not resized properly after removing item from them
**SuperTux version:** latest master
**System information:** *N/A*

##### Expected behavior
Menu is resized properly

##### Actual behavior
Menu is not resized properly. There is empty space left left after deleted item.

##### Steps to reproduce actual behavior
1. Go to Level editor
2. Create levelset and few levels
3. Try deleting them one by one

##### Additional debugging information
Video of the issue: https://github.com/SuperTux/supertux/assets/63350514/5a7c6656-cfa5-451b-bd8a-2f916451d851

### PR DESCRIPTION
**Pull request by mrkubax10**
Fixes #2501

### CODE REVIEW

File: src/gui/menu_item.hpp [hunk 1]
```diff
@@ -22,6 +22,7 @@
 
 class MenuItem
 {
+  friend class Menu;
```
Comment (Vankata453) [inline src/gui/menu_item.hpp:25]: What is this needed for?

Comment (mrkubax10) [reply src/gui/menu_item.hpp:25]: Good catch, I forgot to remove it

File: src/supertux/menu/editor_delete_level_menu.cpp [hunk 2]
```diff
@@ -67,7 +80,10 @@ EditorDeleteLevelMenu::menu_action(MenuItem& item)
       {
         PHYSFS_delete(m_level_full_paths[id].c_str());
         delete_item(id + 2);
-        m_level_full_paths.erase(m_level_full_paths.begin() + id);
+        m_level_full_paths[id].clear();
+        m_level_names[id].clear();
+        clear();
```
Comment (Vankata453) [inline src/supertux/menu/editor_delete_level_menu.cpp:85]: You could add `clear()` at the beginning of `refresh()`.

Comment (mrkubax10) [reply src/supertux/menu/editor_delete_level_menu.cpp:85]: `refresh` is also called in constructor and calling `clear` while menu is empty is redundant.

**Review by Vankata453 — APPROVED** [review]

**Pull request merged**